### PR TITLE
Sam 178 logs implementation

### PR DESCRIPTION
**Pull request by nunomsfranco**
Created log module that will manage the connection with graylog server, through a https connection extending the [severb/graypy](https://github.com/severb/graypy) project and format the messages adding more information to be presented.

Configuration file for Gunicorn setup.

Added possibility to set up some env variables in docker for Redis, VIA and Graylog.

### CODE REVIEW

File: docsbox/docs/views.py [hunk 1]
```diff
@@ -133,16 +154,24 @@ def get(self, task_id):
...
                             abort(504, "VIA service took too long to respond.")
+                    return response
                 else:
+                    app.errlog.log(logging.ERROR, "Task with no result", extra={"request": request, "status": 404})
                     abort(404, message="Task with no result")
             else:
+                app.errlog.log(logging.ERROR, "Task is still queued", extra={"request": request, "status": 400})
                 abort(400, message="Task is still queued")
         else:
+            app.errlog.log(logging.ERROR, "Unknown task_id", extra={"request": request, "status": 404})
```
Comment (numsu) [inline docsbox/docs/views.py]: To handle a bit of the "spam" of these logs could we create our own abort() function, which would handle the error logging and after that call the flask's abort() function?

Comment (nunomsfranco) [reply docsbox/docs/views.py]:
```
def abort(error_code, message, request):
    if error_code >= 500:
        error_level = logging.CRITICAL
    elif error_code >= 400:
        error_level = logging.ERROR
    elif error_code >= 300:
        error_level = logging.WARNING
    app.errlog.log(error_level, message, extra={"request": request, "status": error_code})
    flask_restful.abort(error_code, message)
```

Is this good?

Comment (nunomsfranco) [reply docsbox/docs/views.py]:
```
def abort(error_code, message, request):
    if error_code >= 500:
        error_level = logging.CRITICAL
    elif error_code >= 400:
        error_level = logging.ERROR
    elif error_code >= 300:
        error_level = logging.WARNING
    app.errlog.log(error_level, message, extra={"request": request, "status": error_code})
    flask_abort(error_code, message=message)
```

Is this good? Or can it be improved ?

File: docsbox/docs/views.py [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
 from docsbox.docs.tasks import process_convertion, create_tmp_file_and_get_mimetype, get_task, do_task
 from docsbox.docs.utils import remove_extension, set_options, is_valid_uuid
 from docsbox.docs.via_controller import get_file_from_via  
 
+def abort(error_code, message, request):
+    if error_code >= 500:
+        error_level = logging.CRITICAL
+    elif error_code >= 400:
+        error_level = logging.ERROR
+    elif error_code >= 300:
```
Comment (numsu) [inline docsbox/docs/views.py]: The 3xx HTTP status codes are for redirection. I think we don't need this here. If we want to log a warning we can do it outside the abort method

**Review by numsu — APPROVED** [review]

File: docsbox/docs/views.py (hunk 2 above)
Comment (nunomsfranco) [reply docsbox/docs/views.py]: fixed

**Pull request merged**